### PR TITLE
Abstract the CLI ArgumentParser code into its own module

### PR DESCRIPTION
**Pull request by pdehaan**
r? @rpappalax

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,6 +26,7 @@ test:
 
 clean:
 	rm -rf venv  *egg*  dist  ./docs/_build  .tox _temp
+	mkdir -p _temp/{browsers,profiles}
```
Comment (pdehaan) [inline Makefile]: Make sure the _temp/browsers/ and _temp/profiles directories always exist.

File: fftool/cmd.py [hunk 2]
```diff
@@ -1,124 +0,0 @@
-"""module providing ff-tool command menus"""
```
Comment (pdehaan) [inline fftool/cmd.py]: It seems like this was a dead file.

File: fftool/firefox_download.py [hunk 3]
```diff
@@ -33,15 +33,6 @@
 out = Outlawg()
 
 
-def replace_ext(filename, ext):
```
Comment (pdehaan) [inline fftool/firefox_download.py]: Unused method. :scissors:

File: fftool/firefox_download.py [hunk 4]
```diff
@@ -57,15 +48,16 @@ def download(channel):
 
     download_filename = env.get(channel, 'DOWNLOAD_FILENAME')
     download_path = os.path.join(BASE_DIR, download_filename)
+
+    args = {"channel": channel, "download_path": download_path}
+    print("Downloading {channel} to {download_path}".format(**args))
+
```
Comment (pdehaan) [inline fftool/firefox_download.py]: Moved this up, so I could have the `scraper` code a bit closer together below.

File: fftool/main.py [hunk 5]
```diff
@@ -1,76 +1,23 @@
 #!/usr/bin/env python
 
-import argparse
+from ff_cli import ff_cli
```
Comment (pdehaan) [inline fftool/main.py]: I moved all the ArgumentParser code into a separate module, fftool/ff_cli.py, to try and keep the boilerplate down and not have one "big" 90 line file.

File: fftool/main.py [hunk 6]
```diff
@@ -1,76 +1,23 @@
...
-    options = parser.parse_args()
+    options = ff_cli()
 
     # DOWNLOAD/INSTALL
     download(options.channel)
 
+    # If user specified `--install-only`, then just download/install specified
+    # channel(s) and exit early.
+    if (options.install_only):
+        return
```
Comment (pdehaan) [inline fftool/main.py]: Oh, and added a new flag, `--install-only` (example: `$ ff -c ALL --install-only`), which says "hey, just download/install Firefox, but don't create any profiles or launch any browsers". Conversely, before we would have to do `$ ff -c ALL --no-profile --no-launch` to do the same thing, which was a bit less intuitive.

Comment (rpappalax) [inline fftool/main.py]: Yea, much better!

**Comment by rpappalax** [comment]
:+1:

**Pull request merged**